core/commands/config.go

```diff
@@ -362,29 +398,44 @@ func buildProfileHelp() string {
 	return out
 }
 
-func transformConfig(configRoot string, configName string, transformer config.Transformer) error {
+func transformConfig(configRoot string, configName string, transformer config.Transformer, dryRun bool) (*config.Config, *config.Config, error) {
```

Comment:
It makes more sense to me to just return the `string` with the diff instead of the old and new configs, since the caller has no use for them except to print the diff.

Reply:
Not sure I agree with this. Formatting the diff server-side means the client can't choose how it wants to display it (color, custom diff, etc.). How does this make it cleaner?

Reply:
Both formatting the diff client-side and formatting the diff server-side can solve the problem. If we want to have better flexibility, maybe formatting the diff client-side is a better method.

Reply:
> if, in the future, this needs to be extended to a more configurable command the two jsondiff lines can easily be extracted outside and the old/new config objects can be returned, but I'd wait to see if that need first arises.

My worry with this is that we'd either need to either add an option or change the API. If we do this all client-side, we can add the logic client-side without changing the API at all.

For me, this comes down to the fact these commands are both reasonably stable API endpoints *and* CLI commands. We may never change the output but, really, I can see us deciding that we don't like the JSON diffing algorithm, etc sometime in the future.

Actually, let's get some feedback from the JS people. Really, they should give feedback on the command itself.

Reply:
> For me, this comes down to the fact these commands are both reasonably stable API endpoints _and_ CLI commands. We may never change the output but, really, I can see us deciding that we don't like the JSON diffing algorithm, etc sometime in the future.

Agreed. @chenminjian Could you make that modification  (adding a comment line explainig what the function returns and why) please?

Reply:
Ok. I have push the fixed code.